coalib/results/Diff.py

```diff
@@ -166,16 +166,20 @@ def from_clang_fixit(cls, fixit, file):
         """
         assert isinstance(file, (list, tuple))
 
-        oldvalue = '\n'.join(file[fixit.range.start.line-1:
-                                  fixit.range.end.line])
-        endindex = fixit.range.end.column - len(file[fixit.range.end.line-1])-1
-
-        newvalue = (oldvalue[:fixit.range.start.column-1] +
-                    fixit.value +
-                    oldvalue[endindex:])
-        new_file = (file[:fixit.range.start.line-1] +
-                    type(file)(newvalue.splitlines(True)) +
-                    file[fixit.range.end.line:])
+        try:
+            oldvalue = '\n'.join(file[fixit.range.start.line-1:
+                                      fixit.range.end.line])
+            endindex = fixit.range.end.column - len(file[fixit.range.end.line-1])-1
+
+            newvalue = (oldvalue[:fixit.range.start.column-1] +
+                        fixit.value +
+                        oldvalue[endindex:])
+            new_file = (file[:fixit.range.start.line-1] +
+                        type(file)(newvalue.splitlines(True)) +
+                        file[fixit.range.end.line:])
+        except LibclangError as err:
```

Comment:
Accordingly, after having assert statement in try branch. It could be modified as `except Error as err:` and inside you may handle `LibclangError` separately. This will make the code more robust IMO :grinning:

Reply:
no please keep it. This is exactly the intention of the `except` syntax when you provide a specific exception. Catching broad exceptions is usually a bad thing as other errors that you don't know yet are silently catched, and then the program might go into an unpredicted state (which can cause security issues or whatever...).
Robust code doesn't always mean catching any exception ;)